### ISSUE TITLE
EXCECUTE-Permission on folder not checked when accessing file

### ISSUE THREAD

**Issue by chrhuebner** [marschall/memoryfilesystem#159]
The execute permission on the folder containing the file isn't checked properly.

```java
FileSystem fs = MemoryFileSystemBuilder.newEmpty().addFileAttributeView(PosixFileAttributeView.class).setCaseSensitive(true).build();
final Path path = fs.getPath("/", "sub1", "sub2", "test.txt");
final Path parent = path.getParent();
Files.createDirectories(parent);
Files.write(path, "Hello World".getBytes(), StandardOpenOption.CREATE_NEW);
Files.setPosixFilePermissions(parent, Set.of()); // removes execute permission
final byte[] bytes = Files.readAllBytes(path); // Exception expected
System.out.println(new String(bytes));
```

When I remove the permission on sub1 it works - the exception is thrown:
```java
...
Files.setPosixFilePermissions(parent.getParent(), Set.of()); // removes execute permission on sub1
...
```

The check in https://github.com/marschall/memoryfilesystem/blob/9d9f843e041b7c325ccf416c1e035ece95e2055e/src/main/java/com/github/marschall/memoryfilesystem/MemoryFileSystem.java#L808
should also be done if `isLast` is true. But now for the current directory.

**Comment by chrhuebner** [comment]
It seems as if the execute permission of parent directories are also not checked, when files are deleted.

**Comment by marschall** [comment]
Thanks for reporting, I'll have a look.

**Comment by marschall** [comment]
I released 2.8.1 to Maven Central which should fix the issue.
